app/policies/membership_policy.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class MembershipPolicy < ApplicationPolicy
+  def update?
+    @user.global_administrator? || @user.is_admin_of?(@record.org)
+  end
+
+  def destroy?
+    update? && @record.user.role_level_in(@record.org) != @user.role_level_in(@record.org)
+  end
+
+  def update_global_role?
+    return false unless @user.global_administrator?
```

Comment:
I'd probably write this as a boolean statement rather than as a series of guards, e.g:

```ruby
def update_global_role?
  @user.global_administrator? && target_user_global_level < @user.global_role.level && membership_role_level < @user.global_role.level
end
```

(I think?)

Reply:
Yeah that's pretty hard for me to parse....how about isolating those conditions into descriptive methods. Something like this:

```ruby
def update_global_role?
    @user.global_administrator? && target_user_ranks_lower && target_role_not_above_own_role
  end
```

Reply:
even better